django/utils/http.py

```diff
@@ -297,3 +298,140 @@ def is_safe_url(url, host=None):
         return False
     return ((not url_info.netloc or url_info.netloc == host) and
             (not url_info.scheme or url_info.scheme in ['http', 'https']))
+
+# Functions lifted from their respective modules in order to implement
+# several checks which were not possible with the stdlib
+
+
+def parse_qsl2(qs, keep_blank_values=0, strict_parsing=0, fields_limit=0):
```

Comment:
I think a name something like `limited_parse_qsl_py2` would help distinguish from the built-in version (presumably when we drop Python 2 support, we'll also drop the `_py3` suffix from the one to keep) and also make it clear what the number means.